drivers/memc/memc_mcux_flexspi.c

```diff
@@ -141,10 +144,14 @@ static int memc_flexspi_init(const struct device *dev)
 	}
 
 	/* Apply pinctrl state */
+#ifdef CONFIG_PINCTRL
+	int ret;
```

Comment:
No need to move `int ret;` declaration here if you do `if (IS_ENABLED(CONFIG_PINCTRL)) { ...`

Reply:
We can't used `if (IS_ENABLED(CONFIG_PINCTRL)) {` here, because `data->pincfg` and `PINCTRL_STATE_DEFAULT` are both undeclared if `CONFIG_PINCTRL` is not defined, so compilation of the `pinctrl_apply_state` call will fail.

Reply:
Fair enough.